### PR TITLE
refactor(tree_structured_parzen_estimator): apply static analysis

### PR DESCRIPTION
**Pull request by a-maumau**
## Description

This PR applies some suggestions from the linter to `localization/tree_structured_parzen_estimator`.

Fixed:
- camelCase to snake_case
- use explicit cast
- variable name changed from VAR_NAME to var_name
- [modernize-pass-by-value](https://clang.llvm.org/extra/clang-tidy/checks/modernize/pass-by-value.html)
- (removed unused codes in test)

## Tests performed

Checked with clang-tidy and cppcheck (v2.14.1)
<details>
<summary>check_linter.sh</summary>

```
#!/bin/bash
set -eux

TARGET_DIR=$1

current_dir=$(basename $(pwd))
if [[ ! $current_dir =~ ^(autoware|pilot-auto) ]]; then
    echo "This script must be run in a directory with a prefix of autoware or pilot-auto."
    exit 1
fi

set +eux
export CPLUS_INCLUDE_PATH=/usr/include/c++/11:/usr/include/x86_64-linux-gnu/c++/11:$CPLUS_INCLUDE_PATH
set -eux

fdfind -e cpp -e hpp --full-path ${TARGET_DIR} | xargs -P $(nproc) -I{} cpplint {}
fdfind -e cpp -e hpp --full-path ${TARGET_DIR} | xargs -P $(nproc) -I{} clang-tidy -p build/ {}
```
</details>

---

Before fixing the code:
<details>
<summary>output from above commands</summary>

```Text
$ ./check_linter.sh tree_structured_parzen_estimator
...
3443 warnings generated.
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/include/tree_structured_parzen_estimator/tree_structured_parzen_estimator.hpp:62:3: warning: function 'get_next_input' should be marked [[nodiscard]] [modernize-use-nodiscard]
  Input get_next_input() const;
  ^
  [[nodiscard]] 
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/include/tree_structured_parzen_estimator/tree_structured_parzen_estimator.hpp:65:27: warning: invalid case style for constexpr variable 'MAX_GOOD_RATE' [readability-identifier-naming]
  static constexpr double MAX_GOOD_RATE = 0.10;
                          ^~~~~~~~~~~~~
                          max_good_rate
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/include/tree_structured_parzen_estimator/tree_structured_parzen_estimator.hpp:66:28: warning: invalid case style for constexpr variable 'N_EI_CANDIDATES' [readability-identifier-naming]
  static constexpr int64_t N_EI_CANDIDATES = 100;
                           ^~~~~~~~~~~~~~~
                           n_ei_candidates
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/include/tree_structured_parzen_estimator/tree_structured_parzen_estimator.hpp:70:3: warning: function 'compute_log_likelihood_ratio' should be marked [[nodiscard]] [modernize-use-nodiscard]
  double compute_log_likelihood_ratio(const Input & input) const;
  ^
  [[nodiscard]] 
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/include/tree_structured_parzen_estimator/tree_structured_parzen_estimator.hpp:71:3: warning: function 'log_gaussian_pdf' should be marked [[nodiscard]] [modernize-use-nodiscard]
  double log_gaussian_pdf(const Input & input, const Input & mu, const Input & sigma) const;
  ^
  [[nodiscard]] 
Suppressed 3438 warnings (3438 in non-user code).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
4846 warnings generated.
7713 warnings generated.
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/test/test_tpe.cpp:23:23: warning: narrowing conversion from 'std::vector::size_type' (aka 'unsigned long') to signed type 'int64_t' (aka 'long') is implementation-defined [cppcoreguidelines-narrowing-conversions]
    const int64_t n = input.size();
                      ^
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/test/test_tpe.cpp:31:21: warning: invalid case style for constexpr variable 'kOuterTrialsNum' [readability-identifier-naming]
  constexpr int64_t kOuterTrialsNum = 20;
                    ^~~~~~~~~~~~~~~
                    k_outer_trials_num
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/test/test_tpe.cpp:32:21: warning: invalid case style for constexpr variable 'kInnerTrialsNum' [readability-identifier-naming]
  constexpr int64_t kInnerTrialsNum = 200;
                    ^~~~~~~~~~~~~~~
                    k_inner_trials_num
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/test/test_tpe.cpp:60:31: warning: narrowing conversion from 'std::vector::size_type' (aka 'unsigned long') to 'double' [cppcoreguidelines-narrowing-conversions]
    const double mean = sum / scores.size();
                              ^
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/test/test_tpe.cpp:66:46: warning: narrowing conversion from 'std::vector::size_type' (aka 'unsigned long') to 'double' [cppcoreguidelines-narrowing-conversions]
    const double stddev = std::sqrt(sq_sum / scores.size());
                                             ^
Suppressed 7726 warnings (7708 in non-user code, 18 NOLINT).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
9692 warnings generated.
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/src/tree_structured_parzen_estimator.cpp:27:3: warning: pass by value and use std::move [modernize-pass-by-value]
  const std::vector<double> & sample_mean, const std::vector<double> & sample_stddev)
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~
  std::vector<double> 
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/src/tree_structured_parzen_estimator.cpp:27:44: warning: pass by value and use std::move [modernize-pass-by-value]
  const std::vector<double> & sample_mean, const std::vector<double> & sample_stddev)
                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~
                                           std::vector<double> 
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/src/tree_structured_parzen_estimator.cpp:60:62: warning: narrowing conversion from 'std::vector::size_type' (aka 'unsigned long') to 'double' [cppcoreguidelines-narrowing-conversions]
    std::min({static_cast<int64_t>(10), static_cast<int64_t>(trials_.size() * MAX_GOOD_RATE)});
                                                             ^
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/src/tree_structured_parzen_estimator.cpp:110:21: warning: narrowing conversion from 'std::vector::size_type' (aka 'unsigned long') to signed type 'int64_t' (aka 'long') is implementation-defined [cppcoreguidelines-narrowing-conversions]
  const int64_t n = trials_.size();
                    ^
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/src/tree_structured_parzen_estimator.cpp:120:30: warning: narrowing conversion from 'int64_t' (aka 'long') to 'double' [cppcoreguidelines-narrowing-conversions]
      const double w = 1.0 / above_num_;
                             ^
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/src/tree_structured_parzen_estimator.cpp:124:30: warning: narrowing conversion from 'long' to 'double' [cppcoreguidelines-narrowing-conversions]
      const double w = 1.0 / (n - above_num_);
                             ^
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/src/tree_structured_parzen_estimator.cpp:149:39: warning: method 'log_gaussian_pdf' can be made static [readability-convert-member-functions-to-static]
double TreeStructuredParzenEstimator::log_gaussian_pdf(
                                      ^
/home/masakibaba/autoware/src/universe/autoware.universe/localization/tree_structured_parzen_estimator/src/tree_structured_parzen_estimator.cpp:157:21: warning: narrowing conversion from 'std::vector::size_type' (aka 'unsigned long') to signed type 'int64_t' (aka 'long') is implementation-defined [cppcoreguidelines-narrowing-conversions]
  const int64_t n = input.size();
                    ^
Suppressed 9674 warnings (9674 in non-user code).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
```
</details>

Check with cppcheck:
<details>
<summary>output</summary>

```
$ cppcheck --enable=warning --enable=style --enable=performance --enable=portability --enable=unusedFunction --inconclusive --check-level=exhaustive . -I ../ndt_scan_matcher/src/*

Checking src/tree_structured_parzen_estimator.cpp ...
src/tree_structured_parzen_estimator.cpp:134:11: style: Consider using std::accumulate algorithm instead of a raw loop. [useStlAlgorithm]
      sum += std::exp(log_v - max);
          ^
1/2 files checked 70% done
Checking test/test_tpe.cpp ...
test/test_tpe.cpp:36:49: style: Condition 'n_startup_trials==kInnerTrialsNum' is always false [knownConditionTrueFalse]
  const std::string method = ((n_startup_trials == kInnerTrialsNum) ? "Random" : "TPE");
                                                ^
test/test_tpe.cpp:35:52: note: Assignment 'n_startup_trials=kInnerTrialsNum/10', assigned value is 20
  const int64_t n_startup_trials = kInnerTrialsNum / 10;
                                                   ^
test/test_tpe.cpp:36:49: note: Condition 'n_startup_trials==kInnerTrialsNum' is always false
  const std::string method = ((n_startup_trials == kInnerTrialsNum) ? "Random" : "TPE");
                                                ^
test/test_tpe.cpp:41:22: style: Local variable 'n_startup_trials' shadows outer variable [shadowVariable]
  for (const int64_t n_startup_trials : {kInnerTrialsNum, kInnerTrialsNum / 2}) {
                     ^
test/test_tpe.cpp:35:17: note: Shadowed declaration
  const int64_t n_startup_trials = kInnerTrialsNum / 10;
                ^
test/test_tpe.cpp:41:22: note: Shadow variable
  for (const int64_t n_startup_trials : {kInnerTrialsNum, kInnerTrialsNum / 2}) {
                     ^
test/test_tpe.cpp:42:23: style: Local variable 'method' shadows outer variable [shadowVariable]
    const std::string method = ((n_startup_trials == kInnerTrialsNum) ? "Random" : "TPE");
                      ^
test/test_tpe.cpp:36:21: note: Shadowed declaration
  const std::string method = ((n_startup_trials == kInnerTrialsNum) ? "Random" : "TPE");
                    ^
test/test_tpe.cpp:42:23: note: Shadow variable
    const std::string method = ((n_startup_trials == kInnerTrialsNum) ? "Random" : "TPE");
                      ^
test/test_tpe.cpp:64:14: style: Consider using std::accumulate algorithm instead of a raw loop. [useStlAlgorithm]
      sq_sum += (score - mean) * (score - mean);
             ^
test/test_tpe.cpp:36:21: style: Variable 'method' is assigned a value that is never used. [unreadVariable]
  const std::string method = ((n_startup_trials == kInnerTrialsNum) ? "Random" : "TPE");
                    ^
2/2 files checked 100% done
test/test_tpe.cpp:19:0: style: The function 'TEST' is never used. [unusedFunction]
TEST(TreeStructuredParzenEstimatorTest, TPE_is_better_than_random_search_on_sphere_function)
```
</details>

---

After this PR:

Check with check_linter.sh
```Text
$ ./check_linter.sh tree_structured_parzen_estimator
...
3438 warnings generated.
Suppressed 3438 warnings (3438 in non-user code).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
4832 warnings generated.
7703 warnings generated.
Suppressed 7721 warnings (7703 in non-user code, 18 NOLINT).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
9664 warnings generated.
Suppressed 9664 warnings (9664 in non-user code).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
```

Check with cppcheck:
```
$ cppcheck --enable=warning --enable=style --enable=performance --enable=portability --enable=unusedFunction --inconclusive --check-level=exhaustive . -I ../ndt_scan_matcher/src/*

Checking src/tree_structured_parzen_estimator.cpp ...
1/2 files checked 71% done
Checking test/test_tpe.cpp ...
2/2 files checked 100% done
test/test_tpe.cpp:19:0: style: The function 'TEST' is never used. [unusedFunction]
TEST(TreeStructuredParzenEstimatorTest, TPE_is_better_than_random_search_on_sphere_function)
^
```

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: localization/tree_structured_parzen_estimator/include/tree_structured_parzen_estimator/tree_structured_parzen_estimator.hpp [hunk 1]
```diff
@@ -56,19 +56,20 @@ class TreeStructuredParzenEstimator
 
   TreeStructuredParzenEstimator() = delete;
   TreeStructuredParzenEstimator(
-    const Direction direction, const int64_t n_startup_trials,
-    const std::vector<double> & sample_mean, const std::vector<double> & sample_stddev);
+    const Direction direction, const int64_t n_startup_trials, std::vector<double> & sample_mean,
+    std::vector<double> & sample_stddev);
```
[the remark this answers is not in the document]
Comment (a-maumau) [reply localization/tree_structured_parzen_estimator/include/tree_structured_parzen_estimator/tree_structured_parzen_estimator.hpp]: thank you for mentioning out the point.
I applied your suggestion at 36ea7e0 .

File: localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp [hunk 2]
```diff
@@ -1026,14 +1026,14 @@ geometry_msgs::msg::PoseWithCovarianceStamped NDTScanMatcher::align_pose(
...
-  const std::vector<double> sample_mean{
+  std::vector<double> sample_mean{
     initial_pose_with_cov.pose.pose.position.x,  // trans_x
     initial_pose_with_cov.pose.pose.position.y,  // trans_y
     initial_pose_with_cov.pose.pose.position.z,  // trans_z
     base_rpy.x,                                  // angle_x
     base_rpy.y                                   // angle_y
   };
-  const std::vector<double> sample_stddev{stddev_x, stddev_y, stddev_z, stddev_roll, stddev_pitch};
+  std::vector<double> sample_stddev{stddev_x, stddev_y, stddev_z, stddev_roll, stddev_pitch};
```
Comment (SakodaShintaro) [inline localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp:1036]: Can we restore const in these variables?

**Review by SakodaShintaro — CHANGES_REQUESTED** [review]

File: localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp (hunk 2 above)
Comment (SakodaShintaro) [reply localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp:1036]: Also https://github.com/autowarefoundation/autoware.universe/pull/7364/files#diff-2f9b5f8852197134069202820aee0565d91b6c85c2dc2e5725f3b8f8025cc10fR1029

Comment (a-maumau) [reply localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp]: sorry for the careless commits, I fixed it at 0decbc2 .

File: localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp [hunk 3]
```diff
@@ -1026,7 +1026,7 @@ geometry_msgs::msg::PoseWithCovarianceStamped NDTScanMatcher::align_pose(
   const double stddev_pitch = std::sqrt(covariance(4, 4));
 
   // Since only yaw is uniformly sampled, we define the mean and standard deviation for the others.
-  const std::vector<double> sample_mean{
+  std::vector<double> sample_mean{
```
Comment (SakodaShintaro) [inline localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp:1029]: How about this variable?

Comment (a-maumau) [reply localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp]: thank you again for mentioning out!
I fixed it at ae7b866 .

**Review by SakodaShintaro — APPROVED** [review]
Thank you for the correction!
Looks Good To Me

**Pull request merged**